api/v1/prefixclaim_types.go

```diff
@@ -24,24 +24,31 @@ import (
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // PrefixClaimSpec defines the desired state of PrefixClaim
+// TODO: The reason for using a workaround please see https://github.com/netbox-community/netbox-operator/pull/90#issuecomment-2402112475
 // +kubebuilder:validation:XValidation:rule="!has(oldSelf.site) || has(self.site)", message="Site is required once set"
+// +kubebuilder:validation:XValidation:rule="(!has(self.parentPrefix) && has(self.parentPrefixSelector)) || (has(self.parentPrefix) && !has(self.parentPrefixSelector))"
 type PrefixClaimSpec struct {
+
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	//+kubebuilder:validation:Required
 	//+kubebuilder:validation:Format=cidr
 	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="Field 'parentPrefix' is immutable"
-	ParentPrefix string `json:"parentPrefix"`
+	ParentPrefix string `json:"parentPrefix,omitempty"`
+
+	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="Field 'parentPrefixSelector' is immutable"
+	ParentPrefixSelector map[string]string `json:"parentPrefixSelector,omitempty"`
```

Comment:
Please add as much documentation as possible in here. This way it will appear in the API Schema which will be viewable using e.g. `kubectl explain`. Things such as the following should be mentioned

- You can use both custom fields and built in fields
- For tenant and site use the name instead of the slug

The markdown files should be complementary (e.g. with the example or further reading)

Reply:
Done as requested, also updated the description in `ParentPrefixSelectorGuide.md`.

Reply:
FYI, I have created https://github.com/netbox-community/netbox-operator/issues/128 to track for other fields and CRDs

Reply:
looks good, thanks!